# Problem Statement 3-->
Sales of products in four different regions is tabulated for males and females. Find if male-female buyer rations are similar across regions.

Buyer Ratio.csv

# Business Objectives:-->

To determine whether the ratio of male to female buyers is similar across four different regions (East, West, North, South)

This will help in understanding buyer demographics and assist in regional marketing strategies.

# Constraints:-->
The sample data must be representative of the population.

The chi-square test assumes that expected frequencies should not be too low (generally >5).

In [1]:
#import required libraries
import numpy as np
import pandas as pd
import scipy.stats as stats

In [4]:
#load the dataset
df=pd.read_csv("BuyerRatio.csv")

In [5]:
#description of dataset
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2 entries, 0 to 1
Data columns (total 5 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   Observed Values  2 non-null      object
 1   East             2 non-null      int64 
 2   West             2 non-null      int64 
 3   North            2 non-null      int64 
 4   South            2 non-null      int64 
dtypes: int64(4), object(1)
memory usage: 208.0+ bytes


In [7]:
#print 1st rows to understand the dataset
df.head()

,Observed Values,East,West,North,South
0,Males,50,142,131,70
1,Females,435,1523,1356,750


In [8]:
#shape of dataset
df.shape

(2, 5)

In [9]:
#describe the dataset
df.describe()

,East,West,North,South
count,2.000000,2.000000,2.000000,2.000000
mean,242.500000,832.500000,743.500000,410.000000
std,272.236111,976.514465,866.205807,480.832611
min,50.000000,142.000000,131.000000,70.000000
25%,146.250000,487.250000,437.250000,240.000000
50%,242.500000,832.500000,743.500000,410.000000
75%,338.750000,1177.750000,1049.750000,580.000000
max,435.000000,1523.000000,1356.000000,750.000000


In [10]:
df.columns

Index(['Observed Values', 'East', 'West', 'North', 'South'], dtype='object')

In [12]:
#convert DataFrame to Numpy Array for Analysis
observed = df.iloc[:, 1:].values  # Assuming first column contains labels (Males, Females)
print("Observed Frequencies:\n", observed)

Observed Frequencies:
 [[  50  142  131   70]
 [ 435 1523 1356  750]]


In [13]:
#perform the Chi-Square test
chi2_stat, p_value, dof, expected = stats.chi2_contingency(observed)

In [14]:
# Print results
print("\nChi-Square Statistic:", chi2_stat)
print("p-value:", p_value)
print("Degrees of Freedom:", dof)
print("\nExpected Frequencies (if male-female buyer ratios were similar):\n", pd.DataFrame(expected, index=df.iloc[:, 0], columns=df.columns[1:]))


Chi-Square Statistic: 1.595945538661058
p-value: 0.6603094907091882
Degrees of Freedom: 3

Expected Frequencies (if male-female buyer ratios were similar):
                        East         West        North       South
Observed Values                                                  
Males             42.765313   146.812879   131.117568   72.304241
Females          442.234687  1518.187121  1355.882432  747.695759


Result: p-value = 0.6603

Interpretability:
 
H0: Male-female buyer ratios are the same across all regions.
  
H1: Male-female buyer ratios are different across regions.

Since the p-value (0.6603) is greater than 0.05, we accept the null hypothesis.

Conclusion: Male-female buyer ratios are similar across all regions.